xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2435,17 +2470,61 @@ public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext
 
         // If all else fails, use the default base skin, even if the URLs could be invalid.
         URL url;
+
         if (forceSkinAction) {
             url = urlf.createSkinURL(filename, "skins", getDefaultBaseSkin(context), context);
         } else {
             url = urlf.createSkinURL(filename, getDefaultBaseSkin(context), context);
         }
-        return urlf.getURL(url, context);
+        queryParameters.putAll(getResourceURLParameters(url));
+        return urlf.getURL(url, context, queryParameters);
+    }
+
+    public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext context)
+    {
+        return getSkinFile(filename, forceSkinAction, context, new LinkedHashMap<>());
+    }
+
+    private Map<String, String> getResourceURLParameters(Resource<?> resource)
+    {
+        try {
+            URL resourceUrl = getResource(resource.getPath());
+            return getResourceURLParameters(resourceUrl);
+        } catch (MalformedURLException e) {
+            LOGGER.debug("Error while getting URL for resource filepath [{}]", resource.getPath(), e);
```

Comment:
I would use an error since apparently this is not expected to happen according to the debug message.

Reply:
s/Error while getting URL for resource filepath [{}]/Error while getting URL for resource path [{}]/ ?

Reply:
I'm wondering if it's good to fallback on the version if we have an error. Or if we do then we should log a warning then maybe.